### PR TITLE
Show error message when logging in with unregistered user

### PR DESCRIPTION
**Pull request by MattPutnam**
Also made modals a little easier to create

### CODE REVIEW

File: src/libs/ajax.js [hunk 1]
```diff
@@ -77,8 +77,7 @@ const ajaxService = {
   }
 }
 
-
-const Sam = _.assign({
+export const Sam = _.assign({
```
Comment (zarsky-broad) [inline src/libs/ajax.js]: Two lines before new service definitions

File: src/pages/Main.js [hunk 2]
```diff
@@ -22,13 +24,12 @@ const Main = class Main extends Component {
...
   render() {
-    const { isSignedIn } = this.state
+    const { isSignedIn, isShowingNotRegisteredModal } = this.state
 
     return Fragment([
       div({ id: 'signInButton', style: { display: isSignedIn ? 'none' : 'block' } }),
-      isSignedIn ?
-        this.renderSignedIn() :
-        null
+      isShowingNotRegisteredModal ? this.renderNotRegisteredModal() : null,
```
Comment (zarsky-broad) [inline src/pages/Main.js:34]: A `cond`?

File: js-style.xml [hunk 3]
```diff
@@ -16,6 +16,7 @@
     <option name="SPACES_WITHIN_OBJECT_LITERAL_BRACES" value="true" />
     <option name="SPACES_WITHIN_IMPORTS" value="true" />
     <option name="SPACES_WITHIN_INTERPOLATION_EXPRESSIONS" value="true" />
+    <option name="IMPORT_SORT_MODULE_NAME" value="true" />
```
Comment (zarsky-broad) [inline js-style.xml:19]: I feel like there was some weird result when I tried this before, but that may have been before we were using full paths for our modules.

**Review by zarsky-broad — APPROVED** [review]

File: src/components/Modal.js [hunk 4]
```diff
@@ -71,7 +73,12 @@ export default hh(class Modal extends Component {
                 onClick: onDismiss
               }, 'Cancel') :
               null,
-            okButton
+            Utils.cond(
+              [okButton === undefined, () => buttonPrimary({ onClick: onDismiss }, 'OK')],
+              [_.isString(okButton), () => buttonPrimary({ onClick: onDismiss }, okButton)],
+              [_.isFunction(okButton), () => buttonPrimary({ onClick: okButton }, 'OK')],
+              () => okButton
+            )
```
Comment (panentheos) [inline src/components/Modal.js:82]: Discussion: I appreciate the flexibility offered here, but this seems rather overloaded. Another approach would be keeping the base component simple, and using composition to create additional components that handle specialized cases.

**Review by panentheos — APPROVED** [review]
Looks good.

File: src/components/Modal.js (hunk 4 above)
Comment (zarsky-broad) [reply src/components/Modal.js:82]: Context: this is the pattern from [here](https://github.com/broadinstitute/firecloud-ui/blob/c6d45c846c521ea7c1c984546cb29a11b094c3f8/src/cljs/main/broadfcui/components/modals.cljs#L78)

Comment (dmohs) [reply src/components/Modal.js:82]: +1 to @panentheos's concern. I'd love to see an example of the composition approach, perhaps in a follow-up PR?

File: src/pages/Main.js (hunk 2 above)
Comment (MattPutnam) [reply src/pages/Main.js:34]: No, I potentially want both

**Review by dmohs — APPROVED** [review]

**Pull request merged**